mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -167,28 +197,36 @@ struct RawBufferOpLowering : public ConvertOpToLLVMPattern<GpuOp> {
 
     MemRefDescriptor memrefDescriptor(memref);
 
-    Value ptr = memrefDescriptor.alignedPtr(rewriter, loc);
+    Value ptr = memrefDescriptor.bufferPtr(
```

Comment:
We want `alignedPtr` here - `bufferPtr` in strictly for `malloc()` and `free()`, last I checked

Reply:
No, we want `bufferPtr`, as it adds the offset to the ptr: https://github.com/llvm/llvm-project/blob/main/mlir/lib/Conversion/LLVMCommon/MemRefBuilder.cpp#L191-L213

This was one of the causes of subtle bugs when using a memref with non trivial offset, for example when coming from a subview.

Reply:
I'm pretty sure that these operations aren't meant to work with subviews - and that gEP will be awkwardly non-uniform a bunch.

I think this'll be resolved by moving to addrspace(7) once the LLVM PR for arbitrary types finally lands, then we can just GEP the buffer pointer and not need this stuff.

Reply:
(But good to know what that function does, it seems reasonable in this case)

Reply:
> I'm pretty sure that these operations aren't meant to work with subviews - and that gEP will be awkwardly non-uniform a bunch.

The bug happens in every case with non-trivial offset, whether they come from subviews or not. Here's the bug:

Let's say that offset > 4 and total size of the memref 4. Before, all writes became no-ops and reads 0, because the offset was participating in the index calculation, now that doesn't happen because the offset no longer participates in the index calculation.

Reply:
Hm, yeah, fair, I did try to support this and it is a bug